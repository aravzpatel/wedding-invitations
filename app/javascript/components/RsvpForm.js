import React from "react"
import PropTypes from "prop-types"
import $ from 'jquery';
import Form from 'react-bootstrap/Form'
import Col from 'react-bootstrap/Col'
import Button from 'react-bootstrap/Button'
class RsvPform extends React.Component {
  constructor(props){
    super(props);

    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit(e){
    const data = new FormData(event.target);
    var id = data.get('guestID')
    var firstName = data.get('firstName')
    var lastName = data.get('lastName')
    var email = data.get('email')
    var attending;
    data.get('isGoing') === 'on' ? attending = true : attending = false
    var guestbutton = document.getElementById(id)

    $.ajax({
      url: `/guests`,
      type: 'PUT',
      data: {id: id, first_name: firstName, last_name: lastName, email: email, rsvp: attending},
      success: () => {
        console.log("it worked")
        guestbutton.style.display = "none"
        this.props.onSubmit(); 
      }
    })

    e.preventDefault();
  }

  render () {
    var guests = this.props.guests

    var guest_form = []

    for (const [index, value] of guests.entries()) {
      guest_form.push(
        <div key={value.id} className={value.first_name}>
          <Form onSubmit={this.handleSubmit}>
            <Form.Control type="hidden" name="guestID" value={value.id}></Form.Control>
            <Form.Row>
              <Col>
                <Form.Control name="firstName" defaultValue= {value.first_name}></Form.Control>
              </Col>
              <Col>
                <Form.Control name="lastName" defaultValue= {value.last_name}></Form.Control>
              </Col>
            </Form.Row>
            <br></br>
            <Form.Row>
              <Col>
                <Form.Control name="email" type="email" placeholder="Enter email"></Form.Control>
              </Col>
            </Form.Row>
              
            <br></br>
            <Form.Row className="align-items-center">
              <Col>
                <Form.Check label="Attending?" type="checkbox" name="isGoing"></Form.Check>
                <Form.Control.Feedback type="valid">WooHoo!</Form.Control.Feedback>
              </Col>
              <Col>
                <Button className="rsvp-button" id={value.id} type="submit">
                  Submit
                </Button>
              </Col>
            </Form.Row>
            <br></br>
          </Form>
        </div>
      )
    }

    return (
      <div className="form">
        <div className="header">Please reply by 5th September 2020 to let us know if you can make it</div>
          <div>
            {guest_form}
          </div>
      </div>
    );
  }
}

export default RsvPform
