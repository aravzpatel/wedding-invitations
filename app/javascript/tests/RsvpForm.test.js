import React from 'react';
import RsvpForm from '../components/RsvpForm';
import renderer from 'react-test-renderer';
import $ from 'jquery';
import { shallow } from 'enzyme';


test('Component builds form for one guest', () => {
  const guests = [{id: 1, first_name: 'Jack', last_name: 'Harper'}]
  const component = renderer.create(
    <RsvpForm guests = {guests}/>
  )

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

test('Component builds form for multiple guests', () => {
  const guests = [{id: 1, first_name: 'Jack', last_name: 'Harper'}, {id: 2, first_name: 'Jill', last_name: 'Harper'}]
  const component = renderer.create(
    <RsvpForm guests = {guests}/>
  )

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})


// jest.mock('jquery')
// test('Component calls API when information is filled in', () => {
//   // var $ = require('jquery')
//   const guests = [{id: 1, first_name: 'Jack', last_name: 'Harper'}]
//   const submit = jest.fn()

//   const spy = jest.spyOn(RsvpForm.prototype, 'handleSubmit');

//   const component = shallow(
//     <RsvpForm guests={guests} onSubmit={submit}/>
//   )
//   // const event = {}
//   component.instance().handleSubmit();
//   expect(spy).toHaveBeenCalled();


//   // const spy = jest.spyOn(component.instance(), "handleSubmit");
//   // component.instance().forceUpdate();
  
//   // const button = component.find('.rsvp-button')
//   // console.log(button.debug())

//   // button.click();

//   // console.log(button.debug())
  
//   // // console.log(component.debug())
//   // expect(spy).toHaveBeenCalled()
// })