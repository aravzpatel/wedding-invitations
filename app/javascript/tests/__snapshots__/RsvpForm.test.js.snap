// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component builds form for multiple guests 1`] = `
<div
  className="form"
>
  <div
    className="header"
  >
    Please reply by 30th September 2020 to let us know if you can make it
  </div>
  <div>
    <div
      className="Jack"
    >
      <form
        className=""
        onSubmit={[Function]}
      >
        <input
          className="form-control"
          name="guestID"
          type="hidden"
          value={1}
        />
        <div
          className="form-row"
        >
          <div
            className="col"
          >
            <input
              className="form-control"
              defaultValue="Jack"
              name="firstName"
            />
          </div>
          <br />
          <div
            className="col"
          >
            <input
              className="form-control"
              defaultValue="Harper"
              name="lastName"
            />
          </div>
        </div>
        <br />
        <div
          className="form-row"
        >
          <div
            className="col"
          >
            <input
              className="form-control"
              name="email"
              placeholder="Enter email"
              type="email"
            />
          </div>
        </div>
        <br />
        <div
          className="align-items-center form-row"
        >
          <div
            className="col"
          >
            <div
              className="form-check"
            >
              <input
                className="form-check-input"
                disabled={false}
                name="isGoing"
                type="checkbox"
              />
              <label
                className="form-check-label"
                title=""
              >
                Attending?
              </label>
            </div>
          </div>
          <br />
          <div
            className="col"
          >
            <button
              className="rsvp-button btn btn-primary"
              disabled={false}
              id={1}
              type="submit"
            >
              Submit
            </button>
          </div>
        </div>
        <br />
      </form>
    </div>
    <div
      className="Jill"
    >
      <form
        className=""
        onSubmit={[Function]}
      >
        <input
          className="form-control"
          name="guestID"
          type="hidden"
          value={2}
        />
        <div
          className="form-row"
        >
          <div
            className="col"
          >
            <input
              className="form-control"
              defaultValue="Jill"
              name="firstName"
            />
          </div>
          <br />
          <div
            className="col"
          >
            <input
              className="form-control"
              defaultValue="Harper"
              name="lastName"
            />
          </div>
        </div>
        <br />
        <div
          className="form-row"
        >
          <div
            className="col"
          >
            <input
              className="form-control"
              name="email"
              placeholder="Enter email"
              type="email"
            />
          </div>
        </div>
        <br />
        <div
          className="align-items-center form-row"
        >
          <div
            className="col"
          >
            <div
              className="form-check"
            >
              <input
                className="form-check-input"
                disabled={false}
                name="isGoing"
                type="checkbox"
              />
              <label
                className="form-check-label"
                title=""
              >
                Attending?
              </label>
            </div>
          </div>
          <br />
          <div
            className="col"
          >
            <button
              className="rsvp-button btn btn-primary"
              disabled={false}
              id={2}
              type="submit"
            >
              Submit
            </button>
          </div>
        </div>
        <br />
      </form>
    </div>
  </div>
</div>
`;

exports[`Component builds form for one guest 1`] = `
<div
  className="form"
>
  <div
    className="header"
  >
    Please reply by 30th September 2020 to let us know if you can make it
  </div>
  <div>
    <div
      className="Jack"
    >
      <form
        className=""
        onSubmit={[Function]}
      >
        <input
          className="form-control"
          name="guestID"
          type="hidden"
          value={1}
        />
        <div
          className="form-row"
        >
          <div
            className="col"
          >
            <input
              className="form-control"
              defaultValue="Jack"
              name="firstName"
            />
          </div>
          <br />
          <div
            className="col"
          >
            <input
              className="form-control"
              defaultValue="Harper"
              name="lastName"
            />
          </div>
        </div>
        <br />
        <div
          className="form-row"
        >
          <div
            className="col"
          >
            <input
              className="form-control"
              name="email"
              placeholder="Enter email"
              type="email"
            />
          </div>
        </div>
        <br />
        <div
          className="align-items-center form-row"
        >
          <div
            className="col"
          >
            <div
              className="form-check"
            >
              <input
                className="form-check-input"
                disabled={false}
                name="isGoing"
                type="checkbox"
              />
              <label
                className="form-check-label"
                title=""
              >
                Attending?
              </label>
            </div>
          </div>
          <br />
          <div
            className="col"
          >
            <button
              className="rsvp-button btn btn-primary"
              disabled={false}
              id={1}
              type="submit"
            >
              Submit
            </button>
          </div>
        </div>
        <br />
      </form>
    </div>
  </div>
</div>
`;
